source/languageHandler.py

```diff
@@ -213,76 +319,92 @@ def setLanguage(lang: str) -> None:
-		return _setlocale(category, locale)
-	locale.Error: unsupported locale setting
-	>>> locale.setlocale(locale.LC_ALL, 'en-GB')
-	'en-GB'
-	>>> locale.getlocale()
-	Traceback (most recent call last):
-	File "<stdin>", line 1, in <module>
-	File "Python38-32\lib\locale.py", line 591, in getlocale
-		return _parse_localename(localename)
-	File "Python38-32\lib\locale.py", line 499, in _parse_localename
-		raise ValueError('unknown locale: %s' % localename)
-	ValueError: unknown locale: en-GB
+	Passing NVDA locales straight to python `locale.setlocale` does now work since it tries to normalize the
+	parameter using `locale.normalize` which results in locales unknown to Windows (Python issue 37945).
+	For example executing: `locale.setlocale(locale.LC_ALL, "pl")`
+	results in locale being set to `('pl_PL', 'ISO8859-2')`
+	which is meaningless to Windows,
 	'''
 	originalLocaleName = localeName
-	# Try setting Python's locale to localeName
+	localeString = ""
+	failedToSetLocale = False
 	try:
-		locale.setlocale(locale.LC_ALL, localeName)
-		locale.getlocale()
-		log.debug(f"set python locale to {localeName}")
-		return
-	except locale.Error:
-		log.debugWarning(f"python locale {localeName} could not be set")
+		localeString = localeStringFromLocaleCode(localeName)
+		log.debug(f"Win32 locale string from locale code is {localeString}")
 	except ValueError:
-		log.debugWarning(f"python locale {localeName} could not be retrieved with getlocale")
-
-	if '-' in localeName:
-		# Python couldn't support the language-country locale, try language_country.
+		log.debugWarning(f"Locale {localeName} not supported by Windows")
+	if localeString:
 		try:
-			localeName = localeName.replace('-', '_')
-			locale.setlocale(locale.LC_ALL, localeName)
+			locale.setlocale(locale.LC_ALL, localeString)
 			locale.getlocale()
-			log.debug(f"set python locale to {localeName}")
+			log.debug(f"set python locale to {localeString}")
 			return
 		except locale.Error:
-			log.debugWarning(f"python locale {localeName} could not be set")
+			failedToSetLocale = True
+			log.debugWarning(f"python locale {localeString} could not be set")
 		except ValueError:
-			log.debugWarning(f"python locale {localeName} could not be retrieved with getlocale")
-
-	if '_' in localeName:
-		# Python couldn't support the language_country locale, just try language.
+			failedToSetLocale = True
+			log.debugWarning(f"python locale {localeString} could not be retrieved with getlocale")
+		# The full form langName_country either cannot be retrieved from Windows
+		# or Python cannot be set to that locale.
+		# Try just with the language name.
+		if "_" in localeName:
+			localeName = localeName.split("_")[0]
+			try:
+				localeString = localeStringFromLocaleCode(localeName)
+				log.debug(f"Win32 locale string from locale code is {localeString}")
+			except ValueError:
+				log.debugWarning(f"Locale {localeName} not supported by Windows")

```

Comment:
I think this section can have the indentation reduced, taking it outside of the above `if localeString:` block.

```suggestion
	# The full form langName_country either cannot be retrieved from Windows
	# or Python cannot be set to that locale.
	# Try just with the language name.
	if "_" in localeName:
		localeName = localeName.split("_")[0]
		try:
			localeString = localeStringFromLocaleCode(localeName)
			log.debug(f"Win32 locale string from locale code is {localeString}")
		except ValueError:
			log.debugWarning(f"Locale {localeName} not supported by Windows")
```